src/Middleware/Diagnostics.Abstractions/src/IExceptionHandlerPathFeature.cs

```diff
@@ -13,5 +16,15 @@ public interface IExceptionHandlerPathFeature : IExceptionHandlerFeature
         /// is un-escaped.
         /// </summary>
         string? Path { get; }
+
+        /// <summary>
+        /// Gets the selected <see cref="Http.Endpoint"/> for the original request.
+        /// </summary>
+        Endpoint? Endpoint => null;
+
+        /// <summary>
+        /// Gets the <see cref="RouteValueDictionary"/> associated with the original request.
+        /// </summary>
+        RouteValueDictionary? RouteValues => null;
```

Comment:
If we think there's a chance we'll ever remove IExceptionHandlerPathFeature then we'll want to put these new properties on IExceptionHandlerFeature now. Revisit once we decide what to do with IExceptionHandlerPathFeature.